src/core/meshcop/border_agent.cpp

```diff
@@ -892,6 +894,27 @@ void BorderAgent::HandleDtlsTransportClosed(void)
     RestartAfterRemovingEphemeralKey();
 }
 
+void BorderAgent::SetEphemeralKeyFeatureEnabled(bool aEnabled)
+{
+    VerifyOrExit(mIsEphemeralKeyFeatureEnabled != aEnabled);
+    mIsEphemeralKeyFeatureEnabled = aEnabled;
+
+    if (!mIsEphemeralKeyFeatureEnabled)
+    {
+        // If there is an active session connected with ephemeral key, we disconnect
+        // the session.
+        if (mDidConnectWithEphemeralKey)
```

Comment:
Not sure if using `mDidConnectWithEphemeralKey` is correct here, this is set to true when connected with epskc, but not cleared until next time epskc is set. So let's say there was a connection with epskc and then it was disconnected, later there is a connection with pskc, `mDidConnectWithEphemeralKey` is still true and we will disconnect here. Maybe check both `mDidConnectWithEphemeralKey` and `mUsingEphemeralKey`?

Reply:
Good point! Updated.